pkg/users/client.go

```diff
@@ -238,6 +238,44 @@ type ListAuthFactorsResponse struct {
 	ListMetadata common.ListMetadata `json:"listMetadata"`
 }
 
+type InviteObject struct {
+	Object         string `json:"object"`
+	ID             string `json:"id"`
+	Email          string `json:"email"`
+	State          string `json:"state"`
+	AcceptedAt     string `json:"accepted_at,omitempty"`
+	RevokedAt      string `json:"revoked_at,omitempty"`
+	Token          string `json:"token"`
+	OrganizationID string `json:"organization_id,omitempty"`
+	ExpiresAt      string `json:"expires_at"`
+	CreatedAt      string `json:"created_at"`
+	UpdatedAt      string `json:"updated_at"`
+}
+
+type CreateInvitationOpts struct {
+	Email          string `json:"email"`
+	OrganizationID string `json:"organization_id,omitempty"`
+	ExpiresInDays  int    `json:"expires_in_days,omitempty"`
+	InviterUserID  string `json:"inviter_user_id,omitempty"`
+}
+
+type RevokeInvitationOpts struct {
+	InviteID string
+}
+
+type ListInvitationsOpts struct {
+	OrganizationID string `json:"organization_id,omitempty"`
+	Email          string `json:"email,omitempty"`
+}
+
+type GetInvitationByIDOpts struct {
```

Comment:
```suggestion
type GetInviteOpts struct {
```

looks like the corresponding types for other endpoints don't have a "by ID" suffix (like `GetUserOpts`). let's match that convention; it also matches the API.

Reply:
How does this work when we also have getInvitationByToken? It seems like token and ID should have separate options to pass in? Or are we using a "GetInviteOpts" with a token and ID for both?

Reply:
Or is one `GetInviteOpts` and the other `GetInviteByTokenOpts`?

Reply:
We usually use the same name as the function. So if it's "`GetInvite`" for ID and "`GetInviteByToken`" for token this makes sense. However if we want `GetInviteByID`, structurally we probably want to go with `GetInviteByIDOpts`

Reply:
yeah i was thinking two separate functions `GetInvite` and `GetInviteByToken` (which matches the API implementation). i think as a default we should match the API implementation to keep it straightforward, and then we can make exceptions in cases where the API is inconsistent or not idiomatic. do you think this is one of those cases?

Reply:
Cool as long as it's consistent, I'm 100% for it - will push these changes today.